components/ui/card.tsx

```diff
@@ -0,0 +1,146 @@
+import * as React from 'react'
+import { cn } from '@/lib/utils'
+import { CardI } from '@/lib/interface/controls/card'
+import Image from 'next/image'
+
+const Card = React.forwardRef<HTMLDivElement, CardI>(
+  (
+    {
+      className,
+      title,
+      children,
+      variant = 'default',
+      primaryColor = 'default',
+      imgUrl,
+      members,
+      subject,
+      desc,
+      Icon,
+      ...props
+    },
+    ref
+  ) => (
+    <>
+      {variant === 'default' && (
+        <div
+          ref={ref}
+          className={cn(
+            'rounded-lg border border-slate-200 bg-white  min-w-[280px] max-w-[300px] h-[160px] text-slate-950 shadow-sm dark:border-slate-800 dark:bg-slate-950 dark:text-slate-50',
+            className
+          )}
+          {...props}
+        >
+          <CardHeader
+            className={`${primaryColor == 'default' ? 'bg-[#3A86FF]' : 'bg-[#EECB51]'} font-medium rounded-t-lg flex-row   flex items-center gap-2`}
+          >
+            <div
+              className={`${primaryColor == 'default' ? 'bg-[#66A1FF]' : 'bg-[#FCEBAF]'}
+        text-white px-2 py-2 rounded-tl-lg`}
+            >
+              <Icon />
+            </div>
+            <p>
+              {title.length > 0 && (
+                <CardDescription className="text-white">
+                  {title}
+                </CardDescription>
+              )}
+            </p>
+          </CardHeader>
+          <CardContent>{children}</CardContent>
+        </div>
+      )}
+      {variant === 'image' && (
+        <div
+          className={cn(
+            'rounded-lg border border-slate-200 bg-white  min-w-[300px] pb-[22px] max-w-[320px] min-h-[260px] text-slate-950 shadow-sm dark:border-slate-800 dark:bg-slate-950 dark:text-slate-50',
+            className
+          )}
+          {...props}
+          ref={ref}
+        >
+          <CardHeader className="relative h-[44%] bg-red">
+            <Image
+              src={imgUrl ? imgUrl : ''}
+              alt={title}
+              fill
+              className="object-fit rounded-t-lg"
+            />
+            <div className="absolute bg-[#D3E4FF] px-2 py-1 rounded-lg text-black top-4 right-4">
+              {subject}
+            </div>
+          </CardHeader>
+          <CardContent className="flex flex-col gap-1 pt-[16px] px-[16px] ">
+            <p className="font-[500]">{title}</p>
+            <div className="flex justify-between items-center">
+              <div className="flex items-center gap-3">
+                <div className="w-[30px] h-[30px] rounded-full bg-red-300"></div>
+                <p className="text-sm">Abiel code</p>
+              </div>
+              <p className="text-sm">150 members</p>
+            </div>
+            <div>{desc && <p className='text-sm'>{desc}</p>}  </div>
+          </CardContent>
+        </div>
+      )}
+    </>
+  )
+)
+Card.displayName = 'Card'
+
+const CardHeader = React.forwardRef<
```

Comment:
Extract this into its own file, I recommend a `components` directory in the `components/ui/card` dir, so `components/ui/card/components/header` with the component logic itself defined in `components/ui/card/components/header/index.tsx`.

As stated for the primary `Card` component, use a functional component instead of a `forwardedRef` and define the props type separately, not extending `React.HTMLAttributes<HTMLDivElement>`.

Finally, this component is entirely unnecessary as it simply renders a single div. It should be deleted, and the div included inline in the **Card** component JSX.